src/os_crypto/md5_sha1_sha256/md5_sha1_sha256_op.c

```diff
@@ -61,8 +62,36 @@ int OS_MD5_SHA1_SHA256_File(const char *fname, const char *prefilter_cmd, os_md5
     SHA256_Init(&sha256_ctx);
 
     /* Update for each one */
+#ifdef WIN32
     while ((n = fread(buf, 1, 2048, fp)) > 0) {
+#else
+    while ((n = w_fread_timeout(buf, 1, 2048, fp, 5)) > 0) {
+#endif
+        if (n == 2049) {    // Timeout error
+            mwarn("Timeout when scanning file '%s'. File skipped.", fname);
+            if (prefilter_cmd == NULL) {
+                fclose(fp);
+            } else {
+                pclose(fp);
+            }
+            return (-1);
+        }
+
+        if (max_size > 0) {
+            read = read + n;
+            if (read >= max_size) {     // Maximum filesize error
+                mwarn("Filesize of '%s' is higher than the maximum allowed (%d MB). File skipped.", fname, (int)max_size/1048576); // max_size is in bytes
```

Comment:
`"Filesize of '%s' is higher than the maximum allowed (%d MB)`
>
`"'%s' filesize is larger than the maximum allowed (%d MB)`